src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -27,9 +50,44 @@ public ListPage()
 
     protected override void OnNavigatedTo(NavigationEventArgs e)
     {
+        LoadedState = ViewModelLoadedState.Loading;
         if (e.Parameter is ListViewModel lvm)
         {
-            ViewModel = lvm;
+            if (!lvm.IsInitialized
+                && lvm.InitializeCommand != null)
+            {
+                ViewModel = null;
+                lvm.InitializeCommand.Execute(null);
+
+                _ = Task.Run(async () =>
+                {
+                    await lvm.InitializeCommand.ExecutionTask!;
+
+                    if (lvm.InitializeCommand.ExecutionTask.Status != TaskStatus.RanToCompletion)
+                    {
+                        // TODO: Handle failure case
+                        System.Diagnostics.Debug.WriteLine(lvm.InitializeCommand.ExecutionTask.Exception);
+
+                        _ = _queue.EnqueueAsync(() =>
+                        {
+                            LoadedState = ViewModelLoadedState.Error;
+                        });
```

Comment:
This block seemingly never hits for me if `GetItems` throws an exception (which it can do if the extension threw an exception).

I need to dig in more to see if I'm doing something wrong, or if there's actually a catch that's swallowing it. But this seems like the perfect place to expose some sort of info about the error